### DISCUSSION TITLE
How to migrate a user with pbkdf2 (ASP.NET Core) hashed password?

### DISCUSSION THREAD

**Discussion by aheissenberger** [zitadel/zitadel#8429] [Q&A] [answered]
We use **Zitadel Cloud** and try to import users thru  ImportHumanUser API (v2/users/human). 

It works without password but fails with algorithm "pbkdf2" with error: 
`Password hash encoding not supported.`

**Comment by fforootd** [comment, marked as answer]
Hm can we verify if the encoding of the password is as follows https://github.com/zitadel/passwap?tab=readme-ov-file#pbkdf2

**Reply by aheissenberger** [reply]
Yes, this was the problem. Here is a gist to extract and convert from ASP.NET Core Identity Passwords stored in the database to the Modular Crypt Format.
https://gist.github.com/aheissenberger/fe0b376e56050ffe293f6002956ad8ce

**Reply by fforootd** [reply]
NIce! Thanks for sharing the solution
